invenio_assets/assets/build/webpack.config.js

```diff
@@ -16,14 +17,18 @@ const safePostCssParser = require("postcss-safe-parser");
 const TerserPlugin = require("terser-webpack-plugin");
 const webpack = require("webpack");
 
-// Load aliases from config and resolve their full path
+// Load aliases from config, create missing directories and resolve their full path
 let aliases = {};
 if (config.aliases) {
   aliases = Object.fromEntries(
-    Object.entries(config.aliases).map(
-      ([alias, alias_path]) => [alias, path.resolve(config.build.context, alias_path)]
-    )
-  )
+    Object.entries(config.aliases).map(([alias, alias_path]) => {
+      const resolvedPath = path.resolve(config.build.context, alias_path);
+      if (!fs.existsSync(resolvedPath)) {
```

Comment:
Comment: What is the use case behind the creation of missing aliases paths? Shouldn't be strict instead, in terms of alias path existence?

Reply:
Actually, according to inveniosoftware/invenio-theme#204 I would go probably with the fist solution "Create the folder in the overlay/instance scripts (e.g. cookiecutter, invenio-cli)" or at least restrict the alias path creation to the template only, even if not the cleanest solution...My concern is that the aliases are user defined paths and thus can lead to empty wrong folders and mess up your virtualenv...@slint WDYT?

Reply:
@zzacharo I am not sure I understand how `cookiecutter/invenio-cli` would help on checking the paths folders. Do you have anything in mind?
Can you provide aliases examples? How can you mess up with folders?

IMHO, it should happen every time you run the webpack build, in case you have changed the aliases after the cookiecutter creation, so this solution looks to me the best. It is also not dependent if you are using or not `invenio-cli`.

Reply:
Just trying to wrap my brain around it:

1) The "problem" comes from the dynamic template loading in invenio-theme [templates.js](https://github.com/inveniosoftware/invenio-theme/blob/master/invenio_theme/assets/semantic-ui/js/invenio_theme/templates.js#L15) - i.e. it is a semantic-ui only problem.

2) In general I would be vary about dynamically creating missing aliases folders, as this can mask errors. Example: someone defines a new alias, creates the folder but with a typo - the webpack build will now happily create the folder, and any imports from the alias won't work and the poor developer won't understand anything.

3) The webpack build is for both bootstrap3 and semantic-ui, so I don't think creating a folder only for "@templates" makes sense because it's only needed on semantic-ui.

Isn't it just that cookiecutter-invenio-rdm needs to create a folder ``templates`` inside https://github.com/inveniosoftware/cookiecutter-invenio-instance/tree/master/%7B%7Bcookiecutter.project_shortname%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/theme/assets/semantic-ui ?

Reply:
And a big fat comment inside invenio-theme's ``webpack.py`` that this folder my be created in the cookie-cutter instance?